pkg/services/libraryelements/database.go

```diff
@@ -187,8 +187,18 @@ func (l *LibraryElementService) deleteLibraryElement(c *models.ReqContext, uid s
 	})
 }
 
-// getLibraryElement gets a Library Element.
-func (l *LibraryElementService) getLibraryElement(c *models.ReqContext, uid string) (LibraryElementDTO, error) {
+func selectLibraryElementByParam(params map[string]interface{}) (string, []interface{}) {
+	conditions := make([]string, 0, len(params))
+	values := make([]interface{}, 0, len(params))
+	for param, value := range params {
+		conditions = append(conditions, "le."+param+"=?")
+		values = append(values, value)
+	}
+	return ` WHERE ` + strings.Join(conditions, " AND "), values
+}
+
+// getLibraryElement gets a Library Element where param == value
+func (l *LibraryElementService) getLibraryElement(c *models.ReqContext, params map[string]interface{}) (LibraryElementDTO, error) {
```

Comment:
We need at least one new test for the get by name endpoint. You can add it to the `libraryelements_get_test.go` file.

Reply:
👍